cesium/db.go

```diff
@@ -103,7 +103,7 @@ func (db *DB) Read(ctx context.Context, tr telem.TimeRange, keys ...core.Channel
 // database, a deadlock is caused since the signal context is closed while the writers
 // attempt to send to relay.
 func (db *DB) Close() error {
-	if !db.closed.CompareAndSwap(false, true) {
+	if db.closed.Load() {
```

Comment:
This is sussy. Just loading the closed state here could cause a race condition if close is called concurrently with another DB method

Reply:
resolved by first doing compare and swap then storing false if close failed.

Reply:
Isnt this still an issue? why is this not  a compare and swap rather than storing true at the end. I assume the intention of doing this is so that if a step later on fails, the db isnt still marked as closed. But doesn't this introduce the issue of another thread interpereting the db as open while its in the process of getting closed?

Reply:
No – during the process of the db being closed, the `CompareAndSwap` call makes it so that `closed` is `true` – meaning other operations are not allowed. If the close failed, then those operations are allowed again